### PR TITLE
Use ray on windows as well.

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
#1922 

##### What are the significant changes in functionality due to this change request?
Uses ray on windows.  Ray 1.9 has bugs on windows, so upgrade to 1.12

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 from distutils.command.build import build
 import os
 import sys
+import setuptools
```
Comment (joshua-cogliati-inl) [inline setup.py:18]: Fixes minor bug with setup.py, but not needed by this pull request.

File: dependencies.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@ Note all install methods after "main" take
 -->
 <dependencies>
   <main>
-    <h5py>2.10</h5py>
+    <h5py/>
```
Comment (joshua-cogliati-inl) [inline dependencies.xml:38]: Not strictly needed by this pull request, but raven is not very dependent on the exact version of h5py.

File: dependencies.xml [hunk 3]
```diff
@@ -59,7 +59,8 @@ Note all install methods after "main" take
     <nomkl os='linux' skip_check='True'/>
     <numexpr os='linux'/>
     <cmake skip_check='True' optional='True'/>
-    <ray os="mac,linux" source="pip" pip_extra="[default]">1.9</ray>
+    <ray source="pip" pip_extra="[default]">1.12</ray>
+    <redis source="pip" os='windows'/>
```
Comment (joshua-cogliati-inl) [inline dependencies.xml:63]: For some reason, redis is needed by ray but in windows pip does not get it as a dependency.

File: ravenframework/JobHandler.py [hunk 4]
```diff
@@ -214,7 +215,7 @@ def __initializeRay(self):
           # add names in runInfo
           self.runInfoDict['remoteNodes'] = servers
           ## initialize ray server with nProcs
-          self.rayServer = ray.init(address=address, _redis_password=redisPassword,log_to_driver=False,include_dashboard=db) if _rayAvail else pp.Server(ncpus=int(nProcsHead))
```
Comment (wangcj05) [inline ravenframework/JobHandler.py:217]: could you add an explanation about the remove of redis? it seems ray will no longer launch redis by default.

Comment (joshua-cogliati-inl) [reply ravenframework/JobHandler.py:217]: ray start no longer uses redispassword so I removed it.

Comment (wangcj05) [reply ravenframework/JobHandler.py:217]: In https://docs.ray.io/en/latest/ray-core/package-ref.html, the _redis_password is still present in the method, just by default, ray will not launch redis, am I right here? @joshua-cogliati-inl

Comment (joshua-cogliati-inl) [reply ravenframework/JobHandler.py:217]: Except it is not present in https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-start and ray start --head no longer outputs it.  This change also does not seem to be mentioned in the release notes, so I am at least somewhat mystified, but it does not seem to be used as much any more.

Comment (joshua-cogliati-inl) [reply ravenframework/JobHandler.py:217]: The current usage of redis_password is in: https://github.com/ray-project/ray/blob/master/python/ray/util/collective/collective_group/gloo_collective_group.py#L71

For what it is worth, in ray 1.12, the usage of redis_password is:
```python
            redis_password = ray_constants.REDIS_DEFAULT_PASSWORD
            redisStore.authorize(redis_password)
```
that is, it is completely ignored.

**Review by wangcj05 — APPROVED** [review]
changes look good.

**Comment by wangcj05** [comment]
checklist is good. @joshua-cogliati-inl Could you send out an email regarding the change of raven library dependencies?

**Pull request merged**